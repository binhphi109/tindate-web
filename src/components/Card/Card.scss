.card {
  margin: 16px;
  height: 400px;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  -webkit-box-shadow: 1px 1px 4px 0px rgba(153, 153, 153, 1);
  -moz-box-shadow: 1px 1px 4px 0px rgba(153, 153, 153, 1);
  box-shadow: 1px 1px 4px 0px rgba(153, 153, 153, 1);
  background: #ffffff;

  .cardImage {
    width: 100%;
    flex: 1;
    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;

    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
  }

  .cardFooter {
    height: 42px;
    padding: 16px;
    display: flex;

    .cardTitle {
      font-weight: bold;
      flex: 1;
    }

    .cardSubtitle {
      font-size: 14px;
    }
  }
}
