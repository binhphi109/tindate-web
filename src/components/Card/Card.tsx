import React from "react";
import clsx from "clsx";
import "./Card.scss";

type Props = {
  className?: string;
  imageUrl: string;
  title: string;
  subtitle: string;
};

function Card({ className, imageUrl, title, subtitle }: Props) {
  return (
    <div className={clsx("card", className)}>
      <div
        className="cardImage"
        style={{
          backgroundImage: `url(${imageUrl})`,
        }}
      />
      <div className="cardFooter">
        <div className="cardTitle">{title}</div>
        <div className="cardSubtitle">{subtitle}</div>
      </div>
    </div>
  );
}

Card.defaultProps = {
  className: null,
};

export default Card;
