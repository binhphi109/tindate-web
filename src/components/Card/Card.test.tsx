import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";

describe("components/card/Card", () => {
  test("should match snapshot", () => {
    const wrapper = shallow(<Card image="http://image-url.com" title="Alex" subtitle="25" />);

    expect(wrapper).toMatchSnapshot();
  });
});
