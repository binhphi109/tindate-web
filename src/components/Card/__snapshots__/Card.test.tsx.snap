// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/card/Card should match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="cardImage"
    style={
      Object {
        "backgroundImage": "url(http://image-url.com)",
      }
    }
  />
  <div
    className="cardFooter"
  >
    <div
      className="cardTitle"
    >
      Alex
    </div>
    <div
      className="cardSubtitle"
    >
      25
    </div>
  </div>
</div>
`;
